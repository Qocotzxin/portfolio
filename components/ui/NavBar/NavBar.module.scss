.NavBar {
  width: 100%;
  height: 4rem;
  padding-left: var(--chakra-space-8);
  padding-right: var(--chakra-space-8);
  background: white;
  display: flex;
  justify-content: space-between;
  position: fixed;
  top: 0;
  left: 0;

  &--dark {
    background: var(--chakra-colors-gray-800);
  }

  &__block {
    display: flex;
    align-items: center;
    justify-content: center;

    & > * {
      margin-right: var(--chakra-space-2);
    }

    &--wide {
      & > * {
        margin-right: var(--chakra-space-4);
      }
    }
  }
}
