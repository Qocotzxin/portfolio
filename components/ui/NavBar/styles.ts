import styles from "./NavBar.module.scss";

export const Classes = {
  base: styles.NavBar,
  baseDark: styles["NavBar--dark"],
  block: styles.NavBar__block,
  blockWide: styles["NavBar__block--wide"],
};
