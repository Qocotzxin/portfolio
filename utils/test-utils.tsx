import { ChakraProvider } from "@chakra-ui/react";
import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import theme from "@theme";
import { HygraphProvider } from "contexts/hygraphContext";
import { Components, HygraphData } from "@models/hygraph";

export const hygraphMockData: Partial<HygraphData> = {
  skipLinks: [
    {
      text: "Skip to content",
    },
  ],
  ariaLabels: [
    {
      id: "1",
      metadata: "dark",
      component: Components.ColorModeSwitcher,
      content: "Switch to light mode",
    },
    {
      id: "2",
      metadata: "light",
      component: Components.ColorModeSwitcher,
      content: "Switch to dark mode",
    },
  ],
  navLinks: [
    {
      url: "test",
      order: 1,
      text: "Test",
    },
  ],
  languages: [
    { code: "en", displayName: "english", isActive: true },
    { code: "es", displayName: "spanish", isActive: true },
    { code: "nl", displayName: "dutch", isActive: false },
  ],
  footerLinks: [
    {
      url: "https://github.com",
      isActive: true,
      icon: "github",
      ariaLabel: "Visit Github",
    },
    {
      url: "https://test.com",
      isActive: false,
      icon: "github",
      ariaLabel: "Visit test",
    },
  ],
};

interface AllTheProvidersProps {
  hygraphData?: Partial<HygraphData>;
}

export const AllTheProviders: FC<PropsWithChildren & AllTheProvidersProps> = ({
  children,
  hygraphData = hygraphMockData,
}) => {
  return (
    <ChakraProvider theme={theme}>
      <HygraphProvider hygraphData={hygraphData}>{children}</HygraphProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  wrapperProps?: AllTheProvidersProps
) =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllTheProviders {...wrapperProps}>{children}</AllTheProviders>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { customRender as render };
