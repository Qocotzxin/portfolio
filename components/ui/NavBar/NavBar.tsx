import { useIsDarkMode } from "@hooks/useIsDarkMode";
import { applyConditionalClass, concatClassNames } from "@utils/classnames";
import NextLink from "next/link";
import { FC } from "react";
import ColorModeSwitcher from "./ColorModeSwitcher";
import NavLinks from "./NavLinks";
import NavLogo from "./NavLogo";
import SkipLink from "./SkipLink";
import { Classes } from "./styles";

const NavBar: FC = () => {
  const isDark = useIsDarkMode();

  // TODO: analyze if hiding/showing nav based on scroll is required.
  // const [lastScrollTop, setLastScrollTop] = useState(0);
  // const [navPosition, setNavPosition] = useState<"relative" | "fixed">(
  //   "relative"
  // );

  // const lastScrollTopRef = useRef(lastScrollTop);
  // lastScrollTopRef.current = lastScrollTop;

  // useEffect(() => {
  //   const onScroll = () => {
  //     let scrollTop = window.pageYOffset || document.documentElement.scrollTop;
  //     if (scrollTop > lastScrollTopRef.current) {
  //       setNavPosition("relative");
  //     } else {
  //       setNavPosition("fixed");
  //     }

  //     setLastScrollTop(scrollTop <= 0 ? 0 : scrollTop);
  //   };

  //   window.addEventListener("scroll", onScroll);

  //   return () => {
  //     window.removeEventListener("scroll", onScroll);
  //   };
  // }, []);

  return (
    <nav
      className={concatClassNames(
        Classes.base,
        applyConditionalClass(isDark, Classes.baseDark)
      )}
    >
      <div className={Classes.block}>
        <NextLink href="/" legacyBehavior>
          <a>
            <NavLogo withAnimation />
          </a>
        </NextLink>
      </div>

      <SkipLink />

      <div className={concatClassNames(Classes.block, Classes.blockWide)}>
        <NavLinks />
        <ColorModeSwitcher />
      </div>
    </nav>
  );
};

export default NavBar;
