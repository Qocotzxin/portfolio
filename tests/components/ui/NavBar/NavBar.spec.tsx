import { render } from "@utils/test-utils";
import NavBar from "@ui/NavBar";

describe("NavBar", () => {
  it("Should match snapshot.", () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });
});
