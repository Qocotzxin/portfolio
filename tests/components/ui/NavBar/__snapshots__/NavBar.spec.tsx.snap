// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Should match snapshot. 1`] = `
<div>
  <nav
    class="NavBar NavBar--dark"
  >
    <div
      class="NavBar__block"
    >
      <a
        href="/"
      >
        <svg
          aria-labelledby="title"
          class="NavLogo"
          height="48"
          role="presentation"
          viewBox="0 0 48 48"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title"
            lang="en"
          >
            Logo
          </title>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="gradient"
            x1="0"
            x2="0"
            y1="0"
            y2="100%"
          >
            <stop
              offset="40%"
              stop-color="#4FD1C5"
            />
            <stop
              offset="70%"
              stop-color="#4299E1"
            />
          </lineargradient>
          <circle
            class="NavLogo__outterCircle"
            cx="24"
            cy="24"
            data-testid="NavLogo-outter-circle"
            fill="none"
            r="20"
            stroke="#4FD1C5"
            stroke-dasharray="128"
            stroke-dashoffset="128"
            stroke-width="1"
          />
          <circle
            class="NavLogo__innerCircle"
            cx="24"
            cy="24"
            data-testid="NavLogo-inner-circle"
            fill="none"
            r="16"
            stroke="#4299E1"
            stroke-dasharray="128"
            stroke-dashoffset="128"
            stroke-width="1"
          />
          <text
            class="NavLogo__text fadeIn-regular"
            data-testid="NavLogo-text"
            fill="url(#gradient)"
            x="14"
            y="32"
          >
            CE
          </text>
        </svg>
      </a>
    </div>
    <a
      class="SkipLink"
      data-testid="SkipLink"
      href="#main"
    >
      Skip to content
    </a>
    <div
      class="NavBar__block NavBar__block--wide"
    >
      <ul
        class="NavLinks"
        data-testid="NavLinks-desktop"
      >
        <li>
          <a
            aria-current="false"
            class="Link Link--dark"
            data-order="1"
            href="/test"
            style="order: 1;"
          >
            Test
          </a>
        </li>
      </ul>
      <div
        class="FloatingMenu"
      >
        <div
          class="FloatingMenu"
          data-testid="FloatingMenu"
        >
          <button
            class="Button Button--dark"
            data-testid="NavLinks-mobile"
            id="FloatingMenu-trigger"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16v2H4zm0 5h16v2H4zm0 5h16v2H4z"
              />
            </svg>
          </button>
          <div
            aria-labelledby="FloatingMenu-trigger"
            class="FloatingMenu__list  FloatingMenu__list--dark"
            role="menu"
            tabindex="-1"
          >
            <a
              aria-current="false"
              class="Link Link--dark"
              data-order="1"
              href="/test"
              role="menuitem"
              style="order: 1;"
              tabindex="-1"
            >
              Test
            </a>
          </div>
        </div>
      </div>
      <button
        aria-label="Switch to dark mode"
        aria-live="polite"
        class="ColorModeSwitcher ColorModeSwitcher--dark"
      >
        <div
          class="ColorModeSwitcher__circle ColorModeSwitcher__circle-outter ColorModeSwitcher__circle-outter--dark"
        />
        <div
          class="ColorModeSwitcher__circle ColorModeSwitcher__circle-inner ColorModeSwitcher__circle-inner--dark"
        />
        <svg
          class="ColorModeSwitcher__icon ColorModeSwitcher__icon-stars ColorModeSwitcher__icon-stars--dark"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.657 6.247c.11-.33.576-.33.686 0l.645 1.937a2.89 2.89 0 0 0 1.829 1.828l1.936.645c.33.11.33.576 0 .686l-1.937.645a2.89 2.89 0 0 0-1.828 1.829l-.645 1.936a.361.361 0 0 1-.686 0l-.645-1.937a2.89 2.89 0 0 0-1.828-1.828l-1.937-.645a.361.361 0 0 1 0-.686l1.937-.645a2.89 2.89 0 0 0 1.828-1.828l.645-1.937zM3.794 1.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387A1.734 1.734 0 0 0 4.593 5.69l-.387 1.162a.217.217 0 0 1-.412 0L3.407 5.69A1.734 1.734 0 0 0 2.31 4.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387A1.734 1.734 0 0 0 3.407 2.31l.387-1.162zM10.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732L9.1 2.137a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L10.863.1z"
          />
        </svg>
        <svg
          class="ColorModeSwitcher__icon ColorModeSwitcher__icon-clouds ColorModeSwitcher__icon-clouds--dark"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.473 9a4.5 4.5 0 0 0-8.72-.99A3 3 0 0 0 3 14h8.5a2.5 2.5 0 1 0-.027-5z"
          />
          <path
            d="M14.544 9.772a3.506 3.506 0 0 0-2.225-1.676 5.502 5.502 0 0 0-6.337-4.002 4.002 4.002 0 0 1 7.392.91 2.5 2.5 0 0 1 1.17 4.769z"
          />
        </svg>
      </button>
    </div>
  </nav>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
